src/rgw/rgw_admin.cc

```diff
@@ -1794,19 +1830,38 @@ static void get_md_sync_status(list<string>& status)
     return;
   }
 
+  string local_period = sync_status.sync_info.period;
   map<int, RGWMetadataLogInfo> master_shards_info;
   string master_period;
+  RGWRealm master_realm;
+  string remote,url,access_key,secret_key,realm_id,realm_name;
 
-  ret = sync.read_master_log_shards_info(&master_period, &master_shards_info);
+  /* Set the master zonegroup as the remote */
+  RGWPeriod current_period(local_period);
```

Comment:
the naming of `local_period` and `current_period` here is confusing. the local/current period is already initialized as `RGWRados::current_period`, and can be queried with `store->get_current_period_id()`. the master zone is aggressive about making sure other zones have the latest period, so it's safe to trust our local copy instead of querying it from the master's realm